### PR TITLE
OSSM-1297: Use a remote build cache, if it exists

### PR DESCRIPTION
**Pull request by jwendell**
Manual backport of https://github.com/maistra/envoy/pull/153

### CODE REVIEW

File: maistra/run-ci.sh [hunk 1]
```diff
@@ -15,26 +15,32 @@ export ARCH
...
+    --local_cpu_resources=6 \
+    --jobs=3 \
+    --color=no \
+"
+
+if [ -n "${BAZEL_REMOTE_CACHE}" ]; then
+  COMMON_FLAGS+=" --remote_cache=${BAZEL_REMOTE_CACHE} "
+elif [ -n "${BAZEL_DISK_CACHE}" ]; then
+  COMMON_FLAGS+=" --disk_cache=${BAZEL_DISK_CACHE} "
+fi
```
Comment (tedjpoole) [inline maistra/run-ci.sh:30]: Should there be an `else` clause here so the `--disk_cache=/bazel-cache` option still gets used as the default if neither of the environment variables are set?

Comment (jwendell) [reply maistra/run-ci.sh:30]: I don't think so. We should set the `BAZEL_DISK_CACHE` variable locally if we want to use it. Not that this is already in place for 2.2. This is just a backport to 2.1.

Comment (tedjpoole) [reply maistra/run-ci.sh:30]: I was just thinking about folks who build in a docker container with a `/bazel-cache' mount, and expect it to just work. With this change, the default behaviour will change to using ~/.cache/bazel instead. However, I guess it's simple enough to set the additional environment variable.

**Review by tedjpoole — APPROVED** [review]

File: maistra/run-ci.sh (hunk 1 above)
Comment (jwendell) [reply maistra/run-ci.sh:30]: yeah, that's what I'm doing

File: maistra/run-ci.sh [hunk 2]
```diff
@@ -15,26 +15,32 @@ export ARCH
 export BUILD_SCM_REVISION="Maistra PR #${PULL_NUMBER:-undefined}"
 export BUILD_SCM_STATUS="SHA=${PULL_PULL_SHA:-undefined}"
 
+COMMON_FLAGS="\
+    --incompatible_linkopts_to_linklibs \
+    --local_ram_resources=12288 \
+    --local_cpu_resources=6 \
+    --jobs=3 \
```
Comment (dgn) [inline maistra/run-ci.sh:22]: why the reduction to 3 jobs?

**Review by dgn — APPROVED** [review]

File: maistra/run-ci.sh (hunk 2 above)
Comment (jwendell) [reply maistra/run-ci.sh:22]: we had issues with 4 jobs in 2.2, so, let's reduce everywhere on demand

**Pull request merged**